assert/assertions.go

```diff
@@ -985,6 +985,28 @@ func PanicsWithValue(t TestingT, expected interface{}, f PanicTestFunc, msgAndAr
 	return true
 }
 
+// PanicsWithError asserts that the code inside the specified PanicTestFunc
+// panics, and that the recovered panic value is an error that satisfies the
+// EqualError comparison.
+//
+//   assert.PanicsWithError(t, "crazy error", func(){ GoCrazy() })
+func PanicsWithError(t TestingT, errString string, f PanicTestFunc, msgAndArgs ...interface{}) bool {
+	if h, ok := t.(tHelper); ok {
+		h.Helper()
+	}
+
+	funcDidPanic, panicValue, panickedStack := didPanic(f)
+	if !funcDidPanic {
+		return Fail(t, fmt.Sprintf("func %#v should panic\n\tPanic value:\t%#v", f, panicValue), msgAndArgs...)
+	}
+	panicErr, ok := panicValue.(error)
+	if !ok || !Error(t, panicErr) || !EqualError(t, panicErr, errString) {
+		return Fail(t, fmt.Sprintf("func %#v should panic with error message:\t%#v\n\tPanic value:\t%#v\n\tPanic stack:\t%s", f, errString, panicValue, panickedStack), msgAndArgs...)
```

Comment:
I'm failing a second time here with a more specific error message than that failed with in `Error` or `EqualError`. I'm not sure if there's a better way of doing this. I don't think I can fail twice, and I should probably just duplicate the logic from the other two functions here and return this more specific error. What do you think @georgelesica-wf?

Reply:
Right now it results in two like this:
```
    **********_test.go:20: 
                Error Trace:
                Error:          Error message not equal:
                                expected: "myerror"
                                actual  : "myer ror"
                Test:           TestTest
    **********_test.go:20: 
                Error Trace:
                Error:          func (assert.PanicTestFunc)(0x70b760) should panic with error message:  "myerror"
                                        Panic value:    &errors.errorString{s:"myer ror"}
                                        Panic stack:    goroutine 20 [running]:
...
```

Reply:
I removed the double error in e442a16.